### PR TITLE
Add SegmentedControl component

### PR DESCRIPTION
**Pull request by kulyk**
Adds a radio-like input component (needed for collection types project). Was going to just add one more `variant` to `Radio`, but it also has a vertical display and an option to display default HTML radio buttons, which don't make a lot of sense for the segmented control.

### To Verify
1. Open `_internal/components/segmentedcontrol`
2. Ensure every example makes sense

### Demo

**Default**
default

**Icons**
icons

**Colors**
colors

### CODE REVIEW

File: frontend/src/metabase/components/SegmentedControl.styled.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  padding: 6px 12px;
+
+  border: 1px solid ${color("border")};
+  border-right-width: ${props => (props.isLast ? "1px" : 0)};
+  border-top-left-radius: ${props => (props.isFirst ? BORDER_RADIUS : 0)};
+  border-bottom-left-radius: ${props => (props.isFirst ? BORDER_RADIUS : 0)};
+  border-top-right-radius: ${props => (props.isLast ? BORDER_RADIUS : 0)};
+  border-bottom-right-radius: ${props => (props.isLast ? BORDER_RADIUS : 0)};
+
+  aria-selected: ${props => props.isSelected};
```
Comment (alxnddr) [inline frontend/src/metabase/components/SegmentedControl.styled.js:25]: Not sure if CSS understands `aria` attributes 🤔 
Also it seems like you are using native radio buttons so do we really need to add `aria` attributes? I think voiceover should be working correctly without it, correct me if I'm wrong

**Review by alxnddr — COMMENTED** [review]
Hooray! More reusable components 🎉 
Good job! I left a couple of questions

File: frontend/src/metabase/components/SegmentedControl.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import React, { useMemo } from "react";
+import PropTypes from "prop-types";
+import _ from "underscore";
+import Icon from "metabase/components/Icon";
+import { SegmentedList, SegmentedItem } from "./SegmentedControl.styled";
+
+const optionShape = PropTypes.shape({
+  name: PropTypes.oneOfType([
```
[the remark this answers is not in the document]
Comment (kulyk) [reply frontend/src/metabase/components/SegmentedControl.jsx:8]: Good point, was just adding them one by one and missed the node option

File: frontend/src/metabase/components/SegmentedControl.styled.js (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/components/SegmentedControl.styled.js:25]: Ouch, you're right, I just copied that from `Radio` 😄 Will remove it from there as well

**Review by alxnddr — APPROVED** [review]
👍

**Comment by kulyk** [comment]
@alxnddr, thanks for your review, all comments addressed. Please take a look again once you have time

**Comment by kulyk** [comment]
Wow, that was quick :D Thanks!

**Pull request merged**